File: 53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int res = nums[0];
        int sum = nums[0];
        int n = nums.length;

        for(int i = 1; i < n; ++i){
            sum += nums[i];

            if(res < sum) res = sum;

            if(sum < 0) sum = 0;
        }
        
        return res;
    }
}

File: 53. Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = 0
        sum = 0

        for i in nums:
            if(i < sum + i):
                sum += i
            else: 
                sum = i
            
            if(res < sum):
                res = sum

        return res